endpoints/openrtb2/auction.go

```diff
@@ -249,10 +257,34 @@ func (deps *endpointDeps) Auction(w http.ResponseWriter, r *http.Request, _ http
 		labels, ao = rejectAuctionRequest(*rejectErr, w, deps.hookExecutor, req.BidRequest, account, labels, ao)
 		return
 	}
-
+	setSeatNonBidRaw(req, auctionResponse)
 	labels, ao = sendAuctionResponse(w, deps.hookExecutor, response, req.BidRequest, account, labels, ao)
 }
 
+// setSeatNonBidRaw is temporary introduced function for setting SeatNonBid inside bidResponse.Ext
+// Because,
+// 1. today exchange.HoldAuction prepares and marshals some piece of response.Ext which is then used by auction.go, amp_auction.go and video_auction.go
+// 2. As per discussion with Scott we are planning to move away from - HoldAuction building openrtb2.BidResponse. instead respective auction modules will build this object
+// 3. So, we will need this method to do first,  unmarshalling of response.Ext
+func setSeatNonBidRaw(request *openrtb_ext.RequestWrapper, aucResponse *exchange.AuctionResponse) bool {
+	if aucResponse == nil || aucResponse.BidResponse == nil {
+		return false
+	}
+	// unmarshalling is required here, until we are moving away from bidResponse.Ext, which is populated
+	// by HoldAuction
+	response := aucResponse.BidResponse
+	respExt := new(openrtb_ext.ExtBidResponse)
+
+	if json.Unmarshal(response.Ext, &respExt) == nil && setSeatNonBid(respExt, request, aucResponse) {
+		// marshal again
+		if respExtJson, err := json.Marshal(respExt); err == nil {
+			response.Ext = respExtJson
+			return true
+		}
```

Comment:
I think I'd want to see more error handling here for the unmarshalling/marshalling in the event where the error is not equal to nil. So if either unmarshal/marshal results in an error, I'd want to see that passed upstream rather than it being ignored.

Reply:
Done